### ISSUE TITLE
Yogi serve could be even more awesome!

### ISSUE THREAD

**Issue by solmsted** [yui/yogi#25]
So far my workflow with Yogi has fallen into primarily two stages, first is the command line stage where I frequently build, lint, and occasionally test code.  The second stage is the browser stage with yogi serve where I write api docs, user guide, and occasionally test code.  Sometimes during the second stage I discover something and make a code change and want to re lint and re build.  I have to go back to the command line and kill the server.

It would be super awesome if the yogi serve page had a link to run and display lint output and also have a build button.  Then all of the most common tasks during development could be managed in the browser instead of switching back and forth between the browser and the command line.

Of course there could be a GUI for all of Yogi's functionality, but I think just Lint and Build would be sufficient to streamline the majority of the workflow.

**Comment by davglass** [comment]
What if `yogi serve` also spawned a `shifter --watch` under the hood too?

**Comment by solmsted** [comment]
I like that idea; it could work very well.  I'm not sure if there might be a case where that would cause unexpected builds.  That could either be documented or it could be behind an option like yogi serve --watch.

Could that cause confusion if the build failed for any reason?  Would the build output still write out in the console?  If so, that would probably be sufficient.

So I suppose that turns this into two enhancement requests:
add a shifter --watch feature to yogi serve on the command line
add a way to lint from the yogi server index page

**Comment by davglass** [comment]
I might be able t proxy the `shifter` output to both the console and a web console in the browser. Since `shifter --walk` also can show lint warnings that might work best, no?

**Comment by solmsted** [comment]
That might work, I guess I don't quite understand how the build output would automatically show up in the browser.

**Comment by davglass** [comment]
Since it's a subprocess in node I can trap the output of shifter in flight and have that echoed to the browser. The browser just needs to know that it's to expect it and react to it.

**Comment by davglass** [comment]
You can try this with:

`npm -g install https://github.com/yui/yogi/archive/master.tar.gz`

Then:

```
cd yui3/src/dd
yogi serve --watch --quiet
```

Now `shifter` will `--watch` the current module and build when a change is detected.

**Comment by davglass** [comment]
I could have sworn I commented on this early this morning but it's not showing..

Try the new serve page with this:

`npm -g install https://github.com/yui/yogi/archive/serve.tar.gz`

**Comment by davglass** [comment]
This is now checked in and available as of `yogi@0.047`

**Comment by ignaciocavina** [comment]
This seems not to be working, at least not with newly created modules using yogi init... thou it does worked fine with, for example, yui3/calendar...  

After init'ing a new module and running `yogi serve --watch` I'm getting this:

```
yogi [info]  shifter watching for changes

events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: spawn ENOENT
    at errnoException (child_process.js:975:11)
    at Process.ChildProcess._handle.onexit (child_process.js:766:34)
```
